src/app/containers/ConsentBanner/CanonicalLogic/index.test.js

```diff
@@ -0,0 +1,236 @@
+/* eslint-disable global-require */
+const PRIVACY_COOKIE = 'ckns_privacy';
+const EXPLICIT_COOKIE = 'ckns_explicit';
+const POLICY_COOKIE = 'ckns_policy';
+
+let Cookie;
+let setCookieOvenMock;
+const setShowPrivacyBannerMock = jest.fn();
+const setShowCookieBannerMock = jest.fn();
+
+const setCookieGetMock = ({ privacy = '1', explict = '1', policy = '111' }) => {
+  Cookie.get.mockImplementation(cookie => {
+    if (cookie === PRIVACY_COOKIE) {
+      return privacy;
+    }
+
+    if (cookie === EXPLICIT_COOKIE) {
+      return explict;
+    }
+
+    return policy;
+  });
+};
+
+const getConsentBannerUtilities = ({ logger } = {}) => {
+  const consentBannerUtilities = require('./index').default;
+  return consentBannerUtilities({
+    setShowPrivacyBanner: setShowPrivacyBannerMock,
+    setShowCookieBanner: setShowCookieBannerMock,
+    logger,
+  });
+};
+
+describe('Consent Banner Utilities', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  beforeEach(() => {
+    jest.mock('js-cookie', () => jest.fn());
+    Cookie = require('js-cookie');
+    Cookie.get = jest.fn();
+    Cookie.set = jest.fn();
+
+    jest.mock('./setCookieOven', () => jest.fn());
+    setCookieOvenMock = require('./setCookieOven');
+  });
+
+  describe('runInitial', () => {
+    it('does not show the privacy banner when PRIVACY_COOKIE is 1', () => {
+      setCookieGetMock({ privacy: '1' });
+
+      const { runInitial } = getConsentBannerUtilities();
```

Comment:
Can this line be moved to the `beforeEach`?

We'd have a `left runInitial` up around line 6, which we reset in the `beforeEach`, and we just explicitly call `runInitial` here. I'd also remove the newlines. Should read nice and cleanly then:

```js
      setCookieGetMock({ privacy: '1' });
      runInitial();
      expect(Cookie.set).toHaveBeenCalledTimes(0);
      expect(setShowPrivacyBannerMock).not.toHaveBeenCalled();
```

Reply:
Sadly not, as `setCookieGetMock();` needs to be called before `getConsentBannerUtilities()` as it sets up a mock which needs to happen before the require